lib/rspec/buildkite/analytics/ci.rb

```diff
@@ -2,30 +2,83 @@
 
 require "securerandom"
 
-module RSpec::Buildkite::Analytics::CI
+class RSpec::Buildkite::Analytics::CI
   def self.env
-    if ENV["BUILDKITE_BUILD_ID"]
-      {
-        "CI" => "buildkite",
-        "key" => ENV["BUILDKITE_BUILD_ID"],
-        "url" => ENV["BUILDKITE_BUILD_URL"],
-        "branch" => ENV["BUILDKITE_BRANCH"],
-        "commit_sha" => ENV["BUILDKITE_COMMIT"],
-        "number" => ENV["BUILDKITE_BUILD_NUMBER"],
-        "job_id" => ENV["BUILDKITE_JOB_ID"],
-        "message" => ENV["BUILDKITE_MESSAGE"],
-        "debug" => ENV["BUILDKITE_ANALYTICS_DEBUG_ENABLED"],
-        "version" => RSpec::Buildkite::Analytics::VERSION,
-        "collector" => RSpec::Buildkite::Analytics::NAME
-      }
-    else
-      {
-        "CI" => nil,
-        "key" => SecureRandom.uuid,
-        "debug" => ENV["BUILDKITE_ANALYTICS_DEBUG_ENABLED"],
-        "version" => RSpec::Buildkite::Analytics::VERSION,
-        "collector" => RSpec::Buildkite::Analytics::NAME
-      }
-    end
+    new.env
+  end
+
+  def env
+    ci_env.merge(default)
```

Comment:
I found this a little hard to grok as my initial reaction was "whoa we'll be overwriting the ci specific env with the default!!" and then I read on and you're removing the nil keys so that doesn't happen. But perhaps `default.merge(ci_env)` might be clearer from the get, and you won't have to remove nil keys?

Reply:
I'll rename it. They're not really default, they're more specific. That is, if someone explicitly specifies a `BUILDKITE_ANALYTICS_*` environment variable they're likely wanting to use that value.